pylint/lint/caching.py

```diff
@@ -0,0 +1,65 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+# Copyright (c) https://github.com/PyCQA/pylint/blob/main/CONTRIBUTORS.txt
+
+from __future__ import annotations
+
+import pickle
+import sys
+import warnings
+from pathlib import Path
+
+from pylint.constants import PYLINT_HOME
+from pylint.utils import LinterStats
+
+
+def _get_pdata_path(
+    base_name: Path, recurs: int, pylint_home: Path = Path(PYLINT_HOME)
+) -> Path:
+    underscored_name = "_".join(str(p) for p in base_name.parts)
+    return pylint_home / f"{underscored_name}_{recurs}.stats"
+
+
+def load_results(
+    base: str | Path, pylint_home: str | Path = PYLINT_HOME
+) -> LinterStats | None:
+    base = Path(base)
+    pylint_home = Path(pylint_home)
+    data_file = _get_pdata_path(base, 1, pylint_home)
+
+    if not data_file.exists():
+        return None
+
+    try:
+        with open(data_file, "rb") as stream:
+            data = pickle.load(stream)
+            if not isinstance(data, LinterStats):
+                warnings.warn(
+                    "You're using an old pylint cache with invalid data following "
+                    f"an upgrade, please delete '{data_file}'.",
+                    UserWarning,
+                )
+                raise TypeError
+            return data
+    except TypeError:
+        raise
```

Comment:
```suggestion
                return None
            return data
```
I don't think we need to crash here, it's just a cache miss.